doc/escapable_handle_scope.md

```diff
@@ -0,0 +1,79 @@
+# EscapableHandleScope
+
+The EscapableHandleScope class is used to manage the lifetime of object handles
+which are created through the use of node-addon-api. These handles
+keep an object alive in the heap in order to ensure that the objects
+are not collected while native code is using them.  A handle may be
+created any time a node-addon-api Value(and subclasses) is created
+or returned. An EscapableHandleScope is a special type of HandleScope
```

Comment:
A handle may be created when any new node-addon-api Value and its subclasses have been created or returned.